src/problems/problem_interface.jl

```diff
@@ -0,0 +1,51 @@
+Base.@propagate_inbounds function Base.getindex(prob::AbstractSciMLProblem, sym)
+    if issymbollike(sym)
+        if sym isa AbstractArray
+            return map(s -> prob[s], sym)
+        end
+    end
+
+    if issymbollike(sym)
+        if has_sys(prob.f) && is_indep_sym(prob.f.sys, sym) ||
+            Symbol(sym) == getindepsym(prob)
+            return getindepsym(prob)
+        elseif has_sys(prob.f) && is_param_sym(prob.f.sys, sym)
+            return prob.p[param_sym_to_index(prob.f.sys, sym)]
+        elseif has_paramsyms(prob.f) && Symbol(sym) in getparamsyms(prob)
+            return prob.p[findfirst(x -> isequal(x, Symbol(sym)), getparamsyms(prob))]
+        elseif Symbol(sym) in getsyms(prob)
+            return prob.u0[sym_to_index(sym, prob)]
+        else
+            error("Invalid indexing of problem: $sym is not a state, parameter, or independent variable")
+        end
+    else
+        error("Invalid indexing of problem: $sym is not a symbol")
+    end
+end
+
+function Base.setindex!(prob::AbstractSciMLProblem, val, sym)
+    if has_sys(prob.f)
+        if issymbollike(sym)
+            params = getparamsyms(prob)
+            states = Symbol.(setdiff(getsyms(prob), params))
```

Comment:
why not just use states?

Reply:
symbolic indexing interface wasn't working, and states is an mtk function

Reply:
It's not an MTK function https://github.com/SciML/SymbolicIndexingInterface.jl/blob/master/src/interface.jl#L27

Reply:
if I use states, it can't find the index for some reason